pyiron_contrib/atomistics/atomicrex/structure_list.py

```diff
@@ -418,6 +420,19 @@ def _parse_final_properties(self, struct_lines):
                         len_struct = self._structures._per_chunk_arrays["length"][s_index]
                         final_forces = np.empty((len_struct, 3))
     
+    @property
+    def plot(self):
+        """
+        :class:`.TrainingPlots`: plotting interface
+        """
+        return TrainingPlots(self._structures)
+
+    def prepare_plotting(self, final_values=False):
```

Comment:
Does it make sense to run that automatically in the `plot` property if not done earlier by the user explicitly?  This way they wouldn't need to remember to call this one first.

Reply:
Probably. I just don't know how to treat the final_values then? Right now you can call it using final_values=True. My idea would be to explicitly call prepare_plotting when using final values and within the property when using target values,  but I think it would make sense to agree on some common output handling before that. If i remember correclty you store output in a seperate container and it would be nice to have the interface similar I think

Reply:
I guess one way to do it would to use `FlattenedStorage.split` to have two storages for target and final values separately, but I'm not sure how well that slots with the rest of you setup.

Reply:
I think that would mess up some things, but I am not sure. Also I believe at some point it would be helpful to have target and final values accessible by the plotting class to create plots that show both values/their errors etc.

A clean and flexible but code intensive (in the sense of many lines, not complicated) way would probably be to ask everything that uses plotting to give guaranteed return values.
F.e. whoever wants to use the plot class has to make sure that .cells returns the array of cells .energies returns energies. and so on.
If such guarantees are documented well I guess it would be possible for all of us to have the structure storages we use return correct values using properties, while keeping the way the structures are stored internally completely flexible.
Alternatively we could require the fitting codes to have a to_standard_container() function on their structures and then document how this standard container has to look.

Reply:
As we discussed I'll work on this in a separate PR.